src/download.c

```diff
@@ -289,14 +290,16 @@ int untar_full_download(void *data)
 	}
 
 	/* modern tar will automatically determine the compression type used */
-	string_or_die(&tarcommand, TAR_COMMAND " -C %s/staged/ " TAR_PERM_ATTR_ARGS " -xf %s 2> /dev/null",
-		      state_dir, tarfile);
-
-	err = system(tarcommand);
-	if (WIFEXITED(err)) {
-		err = WEXITSTATUS(err);
-	}
-	free(tarcommand);
+	string_or_die(&tardir, "%s/staged/", state_dir);
+	char *const tarcmd[] = { TAR_COMMAND, "-C", tardir, TAR_PERM_ATTR_ARGS_STRLIST, "-xf", tarfile, NULL };
+	stderrfd = open("/dev/null", O_WRONLY);
+	if (stderrfd == -1) {
```

Comment:
what if this is moved in `system_argv_fd()` for convenience, use something like `fd == -2` to say "use /dev/null"